src/it/scala/temple/containers/PostgresSpec.scala

```diff
@@ -0,0 +1,33 @@
+package temple.containers
+
+import java.sql.{Connection, DriverManager, ResultSet}
+
+import com.spotify.docker.client.DefaultDockerClient
+import com.whisk.docker.DockerFactory
+import com.whisk.docker.impl.spotify.SpotifyDockerFactory
+import com.whisk.docker.scalatest.DockerTestKit
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+
+/** PostgresSpec offers additional methods to test commands using Postgres database */
+trait PostgresSpec extends FlatSpec with DockerTestKit with DockerPostgresService with BeforeAndAfterAll {
+
+  implicit override val dockerFactory: DockerFactory = new SpotifyDockerFactory(DefaultDockerClient.fromEnv().build())
+
+  private def execute[T](fn: Connection => T): Option[T] =
+    Option(
+      DriverManager.getConnection(
+        DockerPostgresService.externalUrl,
+        DockerPostgresService.databaseUsername,
+        DockerPostgresService.databasePassword
+      )
+    ).map { conn =>
+      try fn(conn)
+      finally conn.close()
+    }
```

Comment:
```scala
Option {
  DriverManager.getConnection(
    DockerPostgresService.externalUrl,
    DockerPostgresService.databaseUsername,
    DockerPostgresService.databasePassword
  )
} map { conn =>
  try fn(conn)
  finally conn.close()
}
```
?

Reply:
Not sure whether this is preferable, but the Driver Manager feels like too much of a code block to fit within parens. There’s no hard-and-fast rule of when to use which though